### PR TITLE
Update documentation.

### LINKED ISSUE #56: Improve docs based on @omaciel's comments
https://github.com/SatelliteQE/betelgeuse/pull/53#issuecomment-231204419

### PR DESCRIPTION
**Pull request by sthirugn**
Closes #56, #55

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -61,6 +61,75 @@ Quick Start
...
+Using the example above, Betelgeuse will recognize that there are 2 test cases
+available, and the following attributes will be derived:
+
+* Name: this attribute will be derived from the name of the test method itself
+
+      - test_positive_create_user
+      - test_positive_create_car
+
+* ID: this attribute will be derived from the concatenation of the
+  *module.test_name* or *module.ClassName.test_name* if the test method is
```
Comment (elyezer) [inline docs/index.rst:100]: I think that could be added _in other words the Python import path_.

File: docs/index.rst [hunk 2]
```diff
@@ -61,6 +61,75 @@ Quick Start
...
+  defined within a class. Using our example, the values generated would be:
+
+      - test_user.EntitiesTest.test_positive_create_user
+      - test_user.EntitiesTest.test_positive_create_car
+
+By default, the values automatically derived by Betelgeuse are not very
+flexible, specially in the case when you rename an existing test case or move it
+to a different class or module. It is recommended, therefore, the use of
+testimony `tokens` to provide a bit more information about the tests.
```
Comment (elyezer) [inline docs/index.rst]: s/testimony/Testimony

Comment (elyezer) [inline docs/index.rst]: In RST the inline code is `tokens`

File: docs/index.rst [hunk 3]
```diff
@@ -61,6 +61,75 @@ Quick Start
...
+      import unittest
+
+
+      class EntitiesTest(unittest.TestCase):
+
+          def test_positive_create_user(self):
+          """Create a new user providing all expected attributes.
+
+          @ID: 1d73b8cc-a754-4637-8bae-d9d2aaf89003
+
```
Comment (elyezer) [inline docs/index.rst:122]: Drop this blank line.

File: docs/index.rst [hunk 4]
```diff
@@ -61,6 +61,75 @@ Quick Start
...
+              user = entities.User(name='David', age=20)
+              self.assertEqual(user.name, 'David')
+              self.assertEqual(user.age, 20)
+
+Now Betelgeuse can use this first line to derive a friendlier name for your test
+(instead of using *test_positive_create_user*) and a specific value for its ID.
+Other information can also be added to the docstring to provide more
+information, and this can be handled by the use of Testimony tokens.
+
+.. note:: Make sure that your ``IDs`` are indeed unique per test case.
```
Comment (elyezer) [inline docs/index.rst]: I think we can suggest the usage of uuid4 here as a good ID generator tool.

File: docs/index.rst [hunk 5]
```diff
@@ -214,3 +283,100 @@ With the above report, Betelgeuse performs the following:
     test case in the jUnit XML file.  For this reason, it is highly recommended
     to run ``test-command`` command before ``test-run`` to make sure all
     required test cases are created/updated accordingly.
+
+.. _case_study:
+
+Case Study - A real world sample Test Case
+```````````````````````````````````````````
+Testimony tokens can be used to provide more information about a test case.  The
```
Comment (elyezer) [inline docs/index.rst:307]: Add a blank line above

File: docs/index.rst [hunk 6]
```diff
@@ -61,6 +61,75 @@ Quick Start
 
    .. note:: It is always recommended to use python virtual environment
 
+.. _how_it_works:
+
+How it works?
+`````````````
+Assuming that you have a ``test_user.py`` file with the following content:
```
Comment (elyezer) [inline docs/index.rst:71]: Add a blank line above.

File: docs/index.rst [hunk 7]
```diff
@@ -214,3 +283,100 @@ With the above report, Betelgeuse performs the following:
...
+      @Assert: User is successfully created 
+      @ID: 1d73b8cc-a754-4637-8bae-d9d2aaf89003
+      @Requirement: User Management
+      @CaseAutomation: Automated
+      @CaseLevel: Acceptance
+      @CaseComponent: CLI
+      @TestType: Functional
+      @CaseImportance: High
+      @Upstream: No
+      """
```
Comment (elyezer) [inline docs/index.rst]: The entire docstring block need to be indented

File: docs/index.rst [hunk 8]
```diff
@@ -214,3 +283,100 @@ With the above report, Betelgeuse performs the following:
...
+This feature can be leveraged to minimize the amount of information that needs
+to be written for each test case. Since most of the time, test cases grouped in
+a module usually share the same generic information, one could move most of
+these tokens to the ``module`` level and every single test case found by
+Betelgeuse will inherit these attributes. For example:
+
+
+.. code:: python
+
+    """
```
Comment (elyezer) [inline docs/index.rst]: I will be better to have a first line docstring, something like "Test cases for entities."

File: docs/index.rst [hunk 9]
```diff
@@ -214,3 +283,100 @@ With the above report, Betelgeuse performs the following:
...
+
+    import entities
+    import unittest
+
+
+    class EntitiesTest(unittest.TestCase):
+
+        def test_positive_create_user(self):
+        """Create a new user providing all expected attributes.
```
Comment (elyezer) [inline docs/index.rst]: Indent entire docstring block

File: docs/index.rst [hunk 10]
```diff
@@ -214,3 +283,100 @@ With the above report, Betelgeuse performs the following:
...
+        @Assert: User is successfully created
+        @ID: 1d73b8cc-a754-4637-8bae-d9d2aaf89003
+        """
+            user = entities.User(name='David', age=20)
+            self.assertEqual(user.name, 'David')
+            self.assertEqual(user.age, 20)
+
+
+        def test_positive_create_car(self):
+        """Create a new car providing all expected attributes.
```
Comment (elyezer) [inline docs/index.rst]: Indent entire docstring block

File: docs/index.rst [hunk 11]
```diff
@@ -61,6 +61,75 @@ Quick Start
 
    .. note:: It is always recommended to use python virtual environment
 
+.. _how_it_works:
```
Comment (elyezer) [inline docs/index.rst:66]: Not sure but lines like this one may be breaking the left menu on the docs see the "Usage Examples" and the items below it on the current documentation.

**Comment by elyezer** [comment]
What a in depth documentation. Thank you for adding it and to take care of showing the steps taken in order to have a good documented test module in order to leverage all Betelgeuse and Testimony power.

File: docs/index.rst (hunk 11 above)
Comment (omaciel) [inline docs/index.rst:66]: Can't we generate the docs locally and check if this is an issue?

File: docs/index.rst (hunk 4 above)
Comment (omaciel) [inline docs/index.rst]: We could add a note showing how to do this.

File: docs/index.rst (hunk 7 above)
Comment (omaciel) [inline docs/index.rst]: ~~It does?~~ Ahhh, yes, I see it now :)

File: docs/index.rst (hunk 8 above)
Comment (omaciel) [inline docs/index.rst]: But not a requirement, right?

**Comment by omaciel** [comment]
@sthirugn ACK pending minor changes already suggested by @elyezer

File: docs/index.rst (hunk 8 above)
Comment (sthirugn) [inline docs/index.rst]: Yep, not a requirement, I will add it though.

File: docs/index.rst (hunk 11 above)
Comment (sthirugn) [inline docs/index.rst:66]: betelgeuse-local

Comment (sthirugn) [inline docs/index.rst:66]: It works great locally, it seems to be an issue with how readthedocs processes these specific attributes.

**Comment by sthirugn** [comment]
Thank you for detailed reviews.  All comments updated.

**Pull request merged**